tools/tools.go

```diff
@@ -66,19 +62,22 @@ func DownloadFile(downloadURL, targetDirPath string) error {
 		}
 	}()
 	if err != nil {
-		return fmt.Errorf("failed to create (%s), error: %s", targetDirPath, err)
+		return fmt.Errorf("create %s: %s", targetDirPath, err)
 	}
 
-	resp, err := http.Get(downloadURL)
+	resp, err := retryablehttp.Get(downloadURL)
 	if err != nil {
-		return fmt.Errorf("failed to download from (%s), error: %s", downloadURL, err)
+		return fmt.Errorf("download %s: %s", downloadURL, err)
+	}
+	if resp.StatusCode >= 400 {
```

Comment:
TIL `net/http` follows redirects by default, we should not see 3xx status codes

Reply:
The default retryablehttp config does close http response body https://github.com/hashicorp/go-retryablehttp/blob/9dfd949154e507c91ee569367ffa461991b80950/client.go#L811. closing it twice may not be an issue tough